client/blocks/login/index.jsx

```diff
@@ -3,169 +3,54 @@
  */
 import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
-import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import config from 'config';
-import FormsButton from 'components/forms/form-button';
-import Card from 'components/card';
-import FormPasswordInput from 'components/forms/form-password-input';
-import FormTextInput from 'components/forms/form-text-input';
-import FormCheckbox from 'components/forms/form-checkbox';
-import { loginUser } from 'state/login/actions';
-import Notice from 'components/notice';
 import { createFormAndSubmit } from 'lib/form';
-import { recordTracksEvent } from 'state/analytics/actions';
+import LoginForm from './login-form';
 
-export class Login extends Component {
+class Login extends Component {
 	static propTypes = {
-		loginUser: PropTypes.func.isRequired,
-		translate: PropTypes.func.isRequired,
-		loginError: PropTypes.string,
-		redirectLocation: PropTypes.string,
 		title: PropTypes.string,
+		redirectLocation: PropTypes.string,
+		twoFactorEnabled: PropTypes.bool,
 	};
 
-	static defaultProps = {
-		title: '',
-	};
-
-	state = {
-		usernameOrEmail: '',
-		password: '',
-		rememberme: false,
-		submitting: false,
-		errorMessage: '',
-	};
-
-	dismissNotice = () => {
-		this.setState( {
-			errorMessage: ''
-		} );
-	};
-
-	onChangeField = ( event ) => {
-		const { name, value } = event.target;
-
-		this.setState( { [ name ]: value } );
-	};
-
-	onChangeRememberMe = ( event ) => {
-		const { name, checked } = event.target;
-
-		this.props.recordTracksEvent( 'calypso_login_block_remember_me_change', { new_value: checked } );
-
-		this.setState( { [ name ]: checked } );
-	};
-
-	onSubmitForm = ( event ) => {
-		event.preventDefault();
-		this.setState( {
-			submitting: true
-		} );
-
-		this.props.recordTracksEvent( 'calypso_login_block_login_submit' );
-
-		this.props.loginUser( this.state.usernameOrEmail, this.state.password ).then( () => {
-			this.props.recordTracksEvent( 'calypso_login_block_login_success' );
-			this.dismissNotice();
+	handleValidUsernamePassword = ( { usernameOrEmail, password, redirectLocation, rememberMe } ) => {
```

Comment:
`redirectLocation` is not passed to this function because it's not part of the state of `LoginForm`.